mesa/experimental/components/altair.py

```diff
@@ -34,12 +35,39 @@ def portray(g):
                 all_agent_data.append(agent_data)
         return all_agent_data
 
+    def detect_type(key):
+        key_type = type(all_agent_data[0][key])
+        tooltip_type = ""
+        if key_type == int:
```

Comment:
For checking type, it is recommended with `isinstance`, see https://stackoverflow.com/questions/152580/whats-the-canonical-way-to-check-for-type-in-python:
```
first_by_key = all_agent_data[0][key]
if isinstance(first_by_key, int):
    ...
elif isinstance(first_by_key, ...):
    ...
```